src/ui/components/Library/Team/View/TestRuns/suspense/TestRunsCache.ts

```diff
@@ -12,14 +12,26 @@ import { convertRecording } from "ui/hooks/recordings";
 import { TestGroups, groupRecordings } from "ui/utils/testRuns";
 
 export const testRunsCache = createCache<
-  [graphQLClient: GraphQLClientInterface, accessToken: string | null, workspaceId: string],
+  [
+    graphQLClient: GraphQLClientInterface,
+    accessToken: string | null,
+    workspaceId: string,
+    startTime: string | null,
+    endTime: string | null
+  ],
   TestRun[]
 >({
   config: { immutable: true },
   debugLabel: "testRunsCache",
-  getKey: ([_, __, workspaceId]) => workspaceId,
-  load: async ([graphQLClient, accessToken, workspaceId]) => {
-    const rawTestRuns = await getTestRunsGraphQL(graphQLClient, accessToken, workspaceId);
+  getKey: ([_, __, workspaceId, startTime, endTime]) => `${workspaceId}:${startTime}:${endTime}`,
```

Comment:
endTime will keep on changing. There's no point in using cache. Is there any better pattern?

Reply:
> endTime will keep on changing

What causes the change? Maybe this should be a different _type_ of cache, like an [interval cache](https://suspense.vercel.app/createIntervalCache).

Reply:
Some more background: This `endTime` comes from time filter dropdown. The query it uses accepts `startTime` and `endTime` to filter based on time and get subset of results. Whenever a user switches between "this week" and  "this week", `startTime` remains consistent (it uses -7 or -30 days from current time and sets hour to 0,0). However, `endTime` keeps on changing. It is basically `Date.now()`. I'm using the current timestamp because there maybe new run(s) since last time it was loaded.

I think Interval cache would work. It would be a lot expensive in backend than fetching full range. Thanks!